src/js_sync.hpp

```diff
@@ -39,6 +39,19 @@
 namespace realm {
 namespace js {
 
+static bool config_fs_done = false;
+
+realm::SyncManager& syncManagerShared() {
+    realm::SyncManager& shared = SyncManager::shared();
+    if (!config_fs_done) {
```

Comment:
If you tweak this slightly to be:
```
SyncManager& syncManagerShared() {
    static bool configured = []{
        ensure_directory_exists_for_file(default_realm_file_directory());
        SyncManager::shared().configure_file_system(default_realm_file_directory(), SyncManager::MetadataMode::NoEncryption);
        return true;
    }();
    return SyncManager::shared();
}
```
It'll have the advantage of being thread safe by construction and slightly shorter as well.

Reply:
It also looks like `clear_test_state` still calls `SyncManager::shared()` directly, which may result in problems if `clear_test_state` is called at a point when no sync functionality has otherwise initialized sync.

Reply:
This should either be declared `inline` or have the body of the function moved to a .cpp to avoid the linker errors that CI is currently seeing.

Reply:
Thanks for the suggestion. Also the example is failing on CI for some reason.

Reply:
The CI failure I looked at was due to the lack of `inline` I mentioned above. The js_sync.hpp header ends up included in multiple compilation units so you end up with duplicate copies of the symbol. Marking it as `inline` tells the linker to coalesce them and pick one copy. Moving the definition out of the header prevents there from being duplicates in the first place. Either should address the problem.

Reply:
`clear_test_state` is call before and after every test.